webapp/components/survey/TaxonomyDetails/store/actions/useInit.js

```diff
@@ -1,43 +1,22 @@
 import { useCallback } from 'react'
 import { useParams } from 'react-router'
-import { useDispatch } from 'react-redux'
-
-import axios from 'axios'
 
 import * as A from '@core/arena'
 
-import * as Taxonomy from '@core/survey/taxonomy'
-import { SurveyActions, useSurveyId } from '@webapp/store/survey'
+import * as API from '@webapp/service/api'
+
+import { useSurveyId } from '@webapp/store/survey'
 
 import { State } from '../state'
 
 export const useInit = ({ setState }) => {
   const { taxonomyUuid: taxonomyUuidParam } = useParams()
-  const dispatch = useDispatch()
   const surveyId = useSurveyId()
 
-  return useCallback(async ({ onTaxonomyCreated, taxonomy }) => {
-    let taxonomyToSet = taxonomy
-
-    if (A.isEmpty(taxonomyToSet)) {
-      if (A.isEmpty(taxonomyUuidParam)) {
-        const {
-          data: { taxonomy: taxonomyCreated },
-        } = await axios.post(`/api/survey/${surveyId}/taxonomies`, Taxonomy.newTaxonomy())
-
-        if (onTaxonomyCreated) {
-          onTaxonomyCreated(taxonomyCreated)
-        }
-        taxonomyToSet = taxonomyCreated
-        dispatch(SurveyActions.metaUpdated())
-      } else {
-        const {
-          data: { taxonomy: taxonomyFetched },
-        } = await axios.get(`/api/survey/${surveyId}/taxonomies/${taxonomyUuidParam}?draft=true&validate=true`)
-        taxonomyToSet = taxonomyFetched
-      }
+  return useCallback(async ({ taxonomyUuid }) => {
+    if (!A.isEmpty(taxonomyUuid) || !A.isEmpty(taxonomyUuidParam)) {
```

Comment:
I would still pass only one taxonomyUuid from the component.
And it cannot be empty.
What do you think?

Reply:
solved